docs/tutorial/onchain-rln-relay-chat2.md

```diff
@@ -0,0 +1,242 @@
+```
+You will also see some historical messages  being fetched, again the content may be different on your end:
+
+```
+<Jul 26, 10:41> Bob: hi
+<Jul 26, 10:41> Bob: hi
+<Jun 29, 16:21> Alice: spam1
+<Jun 29, 16:21> Alice: hiiii
+<Jun 29, 16:21> Alice: hello
+<Jun 29, 16:19> Bob: hi
+<Jun 29, 16:19> Bob: hi
+<Jun 29, 16:19> Alice: hi
+<Jun 29, 16:15> b: hi
+<Jun 29, 16:15> h: hi
+...
+```
+Next, you see the following message:
+```
+rln-relay preparation is in progress ...
+```
+At this phase, your rln credentials are getting created and a transaction is being sent to the contract.
+It will take some time for the transaction to be finalized.
+Once finalized, the registered rln identity key, the rln identity commitment key, and the index of the registered credentials will be displayed as given below.
+The rln identity key is not shown in the figure (replaced by a string of `x`s) for security reasons. 
+But, you will see your rln identity key.
+
+```
+your membership index is: 63
+your rln identity key is: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+your rln identity commitment key is: 6c6598126ba10d1b70100893b76d7f8d7343eeb8f5ecfd48371b421c5aa6f012
+```
+
+Finally, the chat prompt `>>` will appear which means your chat2 client is ready.
+Once you type a chat line and hit enter, you will see a message that indicates the epoch at which the message is sent e.g.,
+
+```
+>> Hi
+--rln epoch: 165886530
+<Jul 26, 12:55> Alice: Hi
+```
+The numerical value `165886530` indicates the epoch of the message `Hi`.
+You will see a different value than `165886530` on your screen. 
+If two messages sent by the same chat2 client happen to have the same RLN epoch value, then one of them will be detected as spam and won't be routed (by test fleets in this test setting).
+At the time of this tutorial, the epoch duration is set to `10` seconds.
+You can inspect the current epoch value by checking the following [constant variable](https://github.com/status-im/nim-waku/blob/21cac6d491a6d995a7a8ba84c85fecc7817b3d8b/waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim#L119) in the nim-waku codebase.
+Thus, if you send two messages less than `10` seconds apart, they are likely to get the same `rln epoch` values.
+
+After sending a chat message, you may experience some delay before the next chat prompt appears. 
+The reason is that under the hood a zero-knowledge proof is being generated and attached to your message.
+
+
+Try to spam the network by violating the message rate limit i.e.,
+sending more than one message per epoch. 
+Your messages will be routed via test fleets that are running in spam-protected mode over the same content topic i.e., `/toy-chat/2/luzhou/proto` as your chat client.
+Your samp activity will be detected by them and your message will not reach the rest of the chat clients.
+You can check this by running a second chat user and verifying that spam messages are filtered at the test fleets and not displayed. 
```

Comment:
```suggestion
You can check this by running a second chat user to verify that spam messages are not displayed because are filtered by the test fleets.
```